### PR TITLE
Fix/non faust imports

### PR DESCRIPTION
**Pull request by SRserves85**
Changes:

- created `EmptyFaust` class to handle metaobject imports when `faust` isn't installed.
- using `FaustSerializer` without `faust` now raises Importerror when initialized without `faust`
- can now import MessageSerializer without installing `faust`

### CODE REVIEW

**Comment by SRserves85** [comment]
Thanks for the suggested change. Definitely much more elegant for sure!

**Comment by SRserves85** [comment]
Also it seems you are a huge fan of using avro, and I made a package that essentially does the opposite of what you are achieving with Faust. Essentially taking an avro-schema and creating python classes from it: https://github.com/SRserves85/avro-to-python.

File: schema_registry/serializers/faust_serializer.py [hunk 1]
```diff
@@ -1,76 +1,80 @@
...
-from schema_registry.client import SchemaRegistryClient, schema
+from schema_registry.client import SchemaRegistryClient
+from schema_registry.client.schema import AvroSchema
 from schema_registry.serializers import message_serializer
 
-try:
-    import faust
-except ImportError:  # pragma: no cover
-    faust = None  # type: ignore
+_has_faust = find_spec("faust") is not None
```
Comment (marcosschroh) [inline schema_registry/serializers/faust_serializer.py:8]: Nice!

**Comment by marcosschroh** [comment]
Cool! One last thing. The [CI](https://travis-ci.com/github/marcosschroh/python-schema-registry-client/builds/160915487) is failing because of the linting. You can fix it using `./scripts/lint`.

**Comment by SRserves85** [comment]
Hhmm, I'm unsure why tests are failing now, but let me know how you wish me to fix it.

File: schema_registry/serializers/faust_serializer.py [hunk 2]
```diff
@@ -1,76 +1,74 @@
...
-        }
+_has_faust = find_spec("faust") is not None
 
 
 def serializer_factory(
-    schema_registry_client: SchemaRegistryClient, schema_subject: str, schema: schema.AvroSchema
-) -> Serializer:
-    assert faust is not None, "faust must be installed in order to use FaustSerializer"
+    schema_registry_client: SchemaRegistryClient, schema_subject: str, schema: AvroSchema,
+) -> "Serializer":  # type: ignore
```
Comment (marcosschroh) [inline schema_registry/serializers/faust_serializer.py:13]: Here Flake8 is giving a `false positive`. You can try adding `# noqa: F821` next to the `# type: ignore`.

Comment (SRserves85) [reply schema_registry/serializers/faust_serializer.py]: sweet! Didn't realize I had to add an extra `#` inbetween! thanks a ton and for all the help!

**Pull request merged**